src/main.js

```diff
@@ -25,49 +25,55 @@ export default class LevelSelector extends UICorePlugin {
   bindEvents() {
     this.listenTo(this.core, Events.CORE_READY, this.bindPlaybackEvents)
-    this.listenTo(this.core, Events.CORE_ACTIVE_CONTAINER_CHANGED, this.reload)
+    if (Events.CORE_ACTIVE_CONTAINER_CHANGED)
+      this.listenTo(this.core, Events.CORE_ACTIVE_CONTAINER_CHANGED, this.reload)
+    else
+      this.listenTo(this.core.mediaControl, Events.MEDIACONTROL_CONTAINERCHANGED, this.reload)
     this.listenTo(this.core.mediaControl, Events.MEDIACONTROL_RENDERED, this.render)
     this.listenTo(this.core.mediaControl, Events.MEDIACONTROL_HIDE, this.hideSelectLevelMenu)
   }
 
-  unBindEvents() {
-    this.stopListening(this.core, Events.CORE_READY)
-    this.stopListening(this.core, Events.CORE_ACTIVE_CONTAINER_CHANGED)
-    this.stopListening(this.core.mediaControl, Events.MEDIACONTROL_RENDERED)
-    this.stopListening(this.core.mediaControl, Events.MEDIACONTROL_HIDE)
-    this.stopListening(this.core.activePlayback, Events.PLAYBACK_LEVELS_AVAILABLE)
-    this.stopListening(this.core.activePlayback, Events.PLAYBACK_LEVEL_SWITCH_START)
-    this.stopListening(this.core.activePlayback, Events.PLAYBACK_LEVEL_SWITCH_END)
-    this.stopListening(this.core.activePlayback, Events.PLAYBACK_BITRATE)
-  }
-
   bindPlaybackEvents() {
-    let currentPlayback = this.core.activePlayback
+    if (!this.__playback) return
 
-    this.listenTo(currentPlayback, Events.PLAYBACK_LEVELS_AVAILABLE, this.fillLevels)
-    this.listenTo(currentPlayback, Events.PLAYBACK_LEVEL_SWITCH_START, this.startLevelSwitch)
-    this.listenTo(currentPlayback, Events.PLAYBACK_LEVEL_SWITCH_END, this.stopLevelSwitch)
-    this.listenTo(currentPlayback, Events.PLAYBACK_BITRATE, this.updateCurrentLevel)
+    this.listenTo(this.__playback, Events.PLAYBACK_LEVELS_AVAILABLE, this.fillLevels)
+    this.listenTo(this.__playback, Events.PLAYBACK_LEVEL_SWITCH_START, this.startLevelSwitch)
+    this.listenTo(this.__playback, Events.PLAYBACK_LEVEL_SWITCH_END, this.stopLevelSwitch)
+    this.listenTo(this.__playback, Events.PLAYBACK_BITRATE, this.updateCurrentLevel)
 
-    let playbackLevelsAvaialbeWasTriggered = currentPlayback.levels && currentPlayback.levels.length > 0
-    playbackLevelsAvaialbeWasTriggered && this.fillLevels(currentPlayback.levels)
+    let playbackLevelsAvailableWasTriggered = this.__playback.levels && this.__playback.levels.length > 0
+    playbackLevelsAvailableWasTriggered && this.fillLevels(this.__playback.levels)
   }
 
   reload() {
-    this.unBindEvents()
-    this.bindEvents()
-    this.bindPlaybackEvents()
+    this.stopListening()
+    // Ensure it stop listening before rebind events (avoid duplicate events)
+    process.nextTick(() => {
+      this.bindEvents()
+      this.bindPlaybackEvents()
+    })
   }
 
   shouldRender() {
-    if (!this.core.activeContainer) return false
+    if (!this.__container) return false
 
-    let currentPlayback = this.core.activePlayback
-    if (!currentPlayback) return false
+    if (!this.__playback) return false
```

Comment:
We can check this in one line, what you think?

```suggestion
    if (!this.__playback || !this.__container) return false
```